traveler-ios-ui/TravelerKitUI/ViewControllers/OrderDetailViewController.swift

```diff
@@ -19,16 +19,55 @@ class OrderDetailViewController: UITableViewController {
     @IBOutlet weak var orderDateLabel: UILabel!
     @IBOutlet weak var orderPriceLabel: UILabel!
     @IBOutlet weak var creditCardLabel: UILabel!
+    @IBOutlet weak var emailTicketsButton: UIButton!
+    @IBOutlet weak var cancellationButton: UIButton!
 
-    internal var order: Order?
+    var order: Order?
+
+    private var cancellationQuote: CancellationQuote?
     
     override func viewDidLoad() {
         super.viewDidLoad()
 
-        orderNumberLabel.text = order?.referenceNumber ?? "Order number"
-        orderDateLabel.text = DateFormatter.dateOnlyFormatter.string(from: order!.createdDate)
+        loadOrder()
+    }
+
+    func loadOrder() {
+        orderNumberLabel.text = order?.status == OrderStatus.cancelled ? "Cancelled: \(order?.referenceNumber ?? "")" : order?.referenceNumber
+        orderDateLabel.text = ISO8601DateFormatter.dateOnlyFormatter.string(from: order!.createdDate)
         orderPriceLabel.text = order?.total.localizedDescription
         creditCardLabel.text = "Visa ending in: \(order?.last4Digits ?? "")"
+
+        let title = order?.status == OrderStatus.cancelled ? "View cancellation receipt" : "Cancel order"
+
+        cancellationButton.setTitle(title, for: .normal)
+
+        switch order?.status {
+        case .confirmed?:
+            emailTicketsButton.isEnabled = true
+            cancellationButton.isEnabled = true
+        case .declined?, .pending?, .cancelled?:
+            emailTicketsButton.isEnabled = false
+            cancellationButton.isEnabled = false
+        case .underReview?:
+            emailTicketsButton.isEnabled = true
+            cancellationButton.isEnabled = false
+        case .none:
+            Log("Order has no status", data: nil, level: .error)
```

Comment:
technically this is a "No Order" error because an order always will have a status